File: IT/Strutture dati/Esercizi/Java/Esercizio21.java
/* file Esercizio21.java */

public class Esercizio21 {
	public static void main(String [] args) {
		int [][] matrix = {{1, 2, 3}, {3, 4, 5}, {5, 6, 7}};

		System.out.println("Matrice originale:");

		for (int i = 0; i < matrix.length; i++) {
			for (int j = 0; j < matrix[i].length; j++) {
				System.out.printf("%-2d", matrix[i][j]);
			}
			System.out.print(System.lineSeparator());
		}

		int sc = 3;

		for (int i = 0; i < matrix.length; i++) {
			for (int j = 0; j < matrix[i].length; j++) {
				matrix[i][j] *= sc;
			}
		}

		System.out.println("Matrice moltiplicata per lo scalare " + sc + ":");

		for (int i = 0; i < matrix.length; i++) {
			for (int j = 0; j < matrix[i].length; j++) {
				System.out.printf("%-3d", matrix[i][j]);
			}
			System.out.print(System.lineSeparator());
		}
	}
}

File: IT/Strutture dati/Esercizi/C++/esercizio21.cpp
/* file esercizio21.cpp */

#include <iostream>
#include <iomanip>

using std::cout;
using std::endl;
using std::left;
using std::setw;

int main()
{
	const int M = 3;
	const int N = 3;
	int matrix[M][N] = {{1, 4, 0}, {0, 3, 2}, {4, 6, 9}};

	cout << "Matrice originale: " << endl;

	for (int i = 0; i < 3; i++) {
		for (int j = 0; j < 3; j++) {
			cout << left << setw(2) << matrix[i][j];
		}
		cout << endl;
	}

	int sc = 3;

	for (int i = 0; i < M; i++) {
		for (int j = 0; j < N; j++) {
			matrix[i][j] *= sc;
		}
	}

	cout << "Matrice moltiplicata per lo scalare " << sc << ": " << endl;

	for (int i = 0; i < M; i++) {
		for (int j = 0; j < N; j++) {
			cout << left << setw(3) << matrix[i][j];
		}
		cout << endl;
	}
}

